v2/internal/reconcilers/arm/annotations.go

```diff
@@ -32,3 +35,16 @@ func ClearPollerResumeToken(obj genruntime.MetaObject) {
 	genruntime.RemoveAnnotation(obj, PollerResumeTokenAnnotation)
 	genruntime.RemoveAnnotation(obj, PollerResumeIDAnnotation)
 }
+
+func SetLatestReconciledGeneration(obj genruntime.MetaObject) {
```

Comment:
I thought we could just use the Generation on the status? Does it not work?

Reply:
Ah! it does not. Since we whenever we're polling and polling is not finished, we requeue the resource with `retryAfter` option. Which then requeues the resource and we always set Reconciling condition in `generic_reconciler.go/createOrUpdate` method which updates the observedGeneration on status. Which then is never picked up by the checker we want to add.